### PR TITLE
Installer Detection Enhancements

### PR DESCRIPTION
**Pull request by Trenly**
* Prepare scripts for 1.1.0 manifests
* Correct Null Checks
* When Installer URL is the same for multiple entries, skip re-downloading

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/35735)

### CODE REVIEW

File: Tools/YamlCreate.ps1 [hunk 1]
```diff
@@ -38,8 +38,8 @@ if ($Settings) {
     exit
 }
 
-$ScriptHeader = '# Created with YamlCreate.ps1 v2.0.0'
-$ManifestVersion = '1.0.0'
+$ScriptHeader = '# Created with YamlCreate.ps1 v2.0.2'
```
Comment (vedantmgoyal2009) [inline Tools/YamlCreate.ps1:41]:
```suggestion
$ScriptHeader = '# Created with YamlCreate.ps1 v2.0.1'
```

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]

File: Tools/YamlCreate.ps1 (hunk 1 above)
Comment (Trenly) [reply Tools/YamlCreate.ps1:41]: 2.0.1 is a separate commit. 2.0.2 is appropriate in this case.

(Yes, I know the branch is misnamed)

Comment (vedantmgoyal2009) [reply Tools/YamlCreate.ps1:41]: I thought it was a typo.

**Review by vedantmgoyal9 — DISMISSED** [review]

**Comment by vedantmgoyal2009** [comment]
@denelon

**Review by OfficialEsco — DISMISSED** [review]

**Review by vedantmgoyal9 — APPROVED** [review]

**Review by denelon — APPROVED** [review]

**Pull request merged**